### PR TITLE
ARTEMIS-1676 Support overriding of JAVA_ARGS via env variable

### PR DESCRIPTION
**Pull request by mtaylor**

### CODE REVIEW

File: artemis-cli/src/main/resources/org/apache/activemq/artemis/cli/commands/etc/artemis.profile [hunk 1]
```diff
@@ -21,14 +21,20 @@ ARTEMIS_INSTANCE='${artemis.instance}'
...
 
-# Cluster Properties: Used to pass arguments to ActiveMQ Artemis which can be referenced in broker.xml
-#ARTEMIS_CLUSTER_PROPS="-Dactivemq.remoting.default.port=61617 -Dactivemq.remoting.amqp.port=5673 -Dactivemq.remoting.stomp.port=61614 -Dactivemq.remoting.hornetq.port=5446"
+if [ -z "$ARTEMIS_INSTANCE_URI" ]; then
+    ARTEMIS_INSTANCE_URI='${artemis.instance.uri}'
+fi
 
+# Cluster Properties: Used to pass arguments to ActiveMQ Artemis which can be referenced in broker.xml
+if [ -z "$ARTEMIS_CLUSTER_PROPS" ]; then
+    #ARTEMIS_CLUSTER_PROPS="-Dactivemq.remoting.default.port=61617 -Dactivemq.remoting.amqp.port=5673 -Dactivemq.remoting.stomp.port=61614 -Dactivemq.remoting.hornetq.port=5446"
```
Comment (clebertsuconic) [inline artemis-cli/src/main/resources/org/apache/activemq/artemis/cli/commands/etc/artemis.profile]: what's the point of adding this commented out? the user would have to add it anyways?

**Comment by jbertram** [comment]
I something similar possible for artemis.profile.cmd also?

File: artemis-cli/src/main/resources/org/apache/activemq/artemis/cli/commands/etc/artemis.profile (hunk 1 above)
Comment (mtaylor) [reply artemis-cli/src/main/resources/org/apache/activemq/artemis/cli/commands/etc/artemis.profile]: It is just keeping the same functionality but ensuring the priory isn't overridden when set.  I left in the commented out property as an example

**Comment by mtaylor** [comment]
Yes it's possible. We would need to test it on a windows machine.  That profile is actually structured differently

**Comment by gaohoward** [comment]
@mtaylor can you update the Windows artemis.profile.cmd, I can test it on Windows.

File: artemis-cli/src/main/resources/org/apache/activemq/artemis/cli/commands/etc/artemis.profile (hunk 1 above)
Comment (clebertsuconic) [reply artemis-cli/src/main/resources/org/apache/activemq/artemis/cli/commands/etc/artemis.profile]: I would be ok with this on ARTEMIS_HOME.. ARTEMIS_INSTANCE is a bit dangerous IMO.

Say you went to cd /artemis-servers/server1


and by accident previously you set ARTEMIS-INSTANCE = /artemis-server/myOtherServer


You wouldn't be able to figure out easily... I think it's a bit dangerous...


the best way we can do about this is like what we do now.. we figure out ARTEMIS-INSTANCE by the location of the executable's. if you do /my-other-server/bin/artemis start


then you will know you're starting the right server.

**Comment by mtaylor** [comment]
closing will reopen with suggestions

**Pull request closed without merging**